src/App.tsx

```diff
@@ -1,25 +1,40 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const startIndex = (currentPage - 1) * itemsPerPage + 1;
+  const endIndex = Math.min(startIndex + itemsPerPage - 1, 42);
```

Comment:
The calculation of `startIndex` and `endIndex` is slightly off. The `startIndex` should be `(currentPage - 1) * itemsPerPage` instead of `(currentPage - 1) * itemsPerPage + 1`, as array indices are zero-based. This will ensure that the correct range of items is displayed. Similarly, the `endIndex` should be `startIndex + itemsPerPage` to correctly reflect the number of items per page.